ext/hal/qmsi/drivers/adc/qm_ss_adc.c

```diff
@@ -250,7 +250,10 @@ static void setup_seq_table(const qm_ss_adc_t adc, qm_ss_adc_xfer_t *xfer,
 	/* The sample window is the time in cycles between the start of one
 	 * sample and the start of the next. Resolution is indexed from 0 so we
 	 * need to add 1 and a further 2 for the time it takes to process. */
-	uint16_t delay = (sample_window[adc] - (resolution[adc] + 3));
+	uint16_t delay = 1;
+	if (sample_window[adc] > resolution[adc] + 3) {
+		delay = sample_window[adc] - (resolution[adc] + 3);
+	}
```

Comment:
Style: delay is a stack value here, there's no point in making it 16 bit or unsigned if it's making extra work for you.  The generated instructions are going to be the same either way.  Just write this as:

    int delay = sample_window[adc] - (resolution[adc] + 3);
    delay = delay < 1 ? 1 : delay; /* Clamp underflows */

Then cast to an unsigned short later if needed to stuff it into whatever structure it's going to end up in.